src/qt/paymentserver.cpp

```diff
@@ -414,7 +422,71 @@ void PaymentServer::handleURIOrFile(const QString& s)
 #else
         QUrlQuery uri((QUrl(s)));
 #endif
-        if (uri.hasQueryItem("r")) // payment request URI
+        if (uri.hasQueryItem("m") && uri.hasQueryItem("a")) // payment request URI
+        {
+            QByteArray temp;
+            temp.append(uri.queryItemValue("m"));
+            temp.append(uri.queryItemValue("a"));
+            QString decoded = QUrl::fromPercentEncoding(temp);
+            QUrl fetchUrl(decoded, QUrl::StrictMode);
+
+            if (!model)
+                return;
+
+            CNavCoinAddress addr(uri.queryItemValue("a").toStdString());
+            if (!addr.IsValid())
+            {
+                Q_EMIT message(tr("Verify address"), tr("The entered address is invalid.") + QString(" ") + tr("Please check the address and try again."),
+                               CClientUIInterface::ICON_WARNING);
+                return;
+            }
+            CKeyID keyID;
```

Comment:
What's a key? A Public or Private key? Or something else?

Reply:
In this case, a public key.